src/transformers/configuration_utils.py

```diff
@@ -544,6 +549,20 @@ def from_pretrained(cls, pretrained_model_name_or_path: Union[str, os.PathLike],
         assert config.output_attentions == True
         assert unused_kwargs == {"foo": False}
         ```"""
+        use_auth_token = kwargs.pop("use_auth_token", None)
+        if use_auth_token is not None:
+            warnings.warn(
+                "The `use_auth_token` argument is deprecated and will be removed in v5 of Transformers.", FutureWarning
+            )
+            if token is not None:
+                raise ValueError(
+                    "`token` and `use_auth_token` are both specified. Please set only the argument `token`."
+                )
+            token = use_auth_token
+
+            # change to `token` in a follow-up PR
+            kwargs["use_auth_token"] = token
```

Comment:
The whole `kwargs` is used later in this method.

Reply:
This needs to be done if `token` is not `None` outside of this test.